lib/streams.js

```diff
@@ -87,7 +87,7 @@ class BlobReadStream extends Readable {
   constructor (core, id, opts = {}) {
     super(opts)
     this.id = id
-    this.core = core
+    this.core = core.snapshot()
 
     this._controller = new AbortController()
```

Comment:
you don't need to abort controller now

Reply:
and add `_predestroy () { this.core.close().then(noop, noop) }` for perfect error flow (so .destroy() from the outside cancels a pending request)

Reply:
K added